web/src/screens/private/Events/EditEvent.js

```diff
@@ -131,8 +453,46 @@ const userQuery = graphql(CURRENT_USER_QUERY, {
   }),
 });
 
+const createEvent = graphql(CREATE_EVENT_MUTATION, {
+  props: ({ mutate }) => ({
+    createEvent: ({ name, details, sourceIdentifier, permalink, eventLocations, groupId }) =>
+      mutate({
+        variables: {
+          event: { name, details, sourceIdentifier, permalink, eventLocations, groupId },
+        },
+        refetchQueries: [
+          {
+            query: CURRENT_USER_QUERY,
+          },
+        ],
+      }),
+  }),
+});
+
+const updateEvent = graphql(UPDATE_EVENT_MUTATION, {
+  props: ({ mutate }) => ({
+    updateEvent: ({ id, name, details, sourceIdentifier, permalink, eventLocations, groupId }) =>
+      mutate({
+        variables: {
+          event: { id, name, details, sourceIdentifier, permalink, eventLocations, groupId },
+        },
+        refetchQueries: [
+          {
+            query: CURRENT_USER_QUERY,
+          },
+        ],
+      }),
+  }),
+});
+
 const mapStateToProps = ({ auth }) => ({
   auth,
 });
 
-export default compose(connect(mapStateToProps), withStyles(styles), userQuery)(EditEvent);
+export default compose(
+  connect(mapStateToProps),
+  withStyles(styles),
+  createEvent,
+  updateEvent,
+  userQuery,
+)(EditEvent);
```

Comment:
this component is massive - in future I think we should try to break it up. not now :)

Reply:
Yes there are definitely parts that can be moved to components.